<a href="https://colab.research.google.com/github/vinihora/portfolio_data_science/blob/main/projects/covid19_desenvolvidos_vs_subdesenvolvidos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img alt="Colaboratory logo" width="15%" src="https://raw.githubusercontent.com/carlosfab/escola-data-science/master/img/novo_logo_bg_claro.png">

#### **Data Science na Prática 3.0**
*by [sigmoidal.ai](https://sigmoidal.ai)*  

---
  

# **Panorama do COVID-19 no Brasil**

COVID-19 é uma doença infecciosa causada por recém-descoberto coronavírus.

Transmitido principalmente por meio de gotículas provenientes de tosses ou espirros de pessoas infectadas, a gravidade dos sintomas varia muito de pessoa para pessoa.

Fato é, não se sabe muita coisa a respeito do COVID-19. Estudos estão sendo realizados no mundo todo, porém os resultados ainda não são conclusivos e definitivos.

<center><img alt="Colaboratory logo" width="50%" src="https://raw.githubusercontent.com/carlosfab/dsnp2/master/img/covid19.jpg"></center>

Até o presente momento, observa-se que cerca de 80% dos casos confirmados são assintomáticos e rápidos. A maioria das pessoas que se encaixam nesse grupo, se recupera sem nenhuma sequela.

No entanto, 15% das pessoas terão infecções graves e precisarão de oxigênio. O restante das pessoas, que representam 5%, serão classificadas como infecções muito graves e precisarão de ventilação assistida, por meio de respiradores mecânicos em ambiente hospitalar.

# **Objetivos**

Com o objetivo de inferir os resultados de COVID ao redor do mundo, realizarei uma análise voltada ao impacto do vírus em países desenvolvidos e subdesenvolvidos.

Mesmo após o fim daquilo que era considerado como uma pandemia, será que o vírus ainda é uma extrema ameaças para áqueles a margem da economia mundial? O que os dados podem nos revelar sobre ?

É inegável que a pandemia trouxe modificações permanentes a toda a todo o mundo, independe de sua situação financeira. Contudo, será que o seu dano foi proporcional a todas as economias?

Durante o estudo, buscaremos informações a partir dos dados que informe o nível de impacto do COVID em mortes e desenvolvimento econômico.

## **1. Obtenção dos Dados**

* Os dados foram coletados do repositório público de COVID-19 mantido pela 'Our World in Data'.
* A base é atualizada diariamente no caso de ocorrência de novos casos em determinada região.
* Apesar da manutenção dos dados ser realizada por fontes confiáveis como a Our World in Data, existe uma enorme dificuldade em sua coleta, devido as inúmeras fontes que muitas vezes divergem em suas informaçoes, principalmente no ínicio do perído, que não existiam testes suficientes para o acompanhamento em real-time dos casos confirmados.



### Dicionário de Variáveis
- ```iso_code``` - ISO 3166 - Código de 3 letras que representam os países
- ```continent``` - Continente o qual o país se localiza
- ```location``` - Localização geográfica
- ```date``` - Data da observação
- ```total_cases``` - Total confirmado de casos de COVID-19, a depender da fonte, pode incluir casos prováveis
- ```new_cases``` - Novos casos confirmados
- ```new_cases_smoothed``` - Novos casos (Média dos últimos 7 dias), incluindo o dia atual
- ```total_deaths``` - Número total de mortes pelo COVID-19
- ```new_deaths``` - Novas mortes pelo COVID-19
- ```new_deaths_smoothed``` - Novas mortes pelo COVID-10 (Média dos últimos 7 dias), incluindo o dia atual
- ```total_cases_per_million``` - Número total de casos a cada 1.000.000 de pessoas
- ```new_cases_per_million``` - Novos casos de COVID-19 a cada 1.000.000 de pessoas
- ```new_cases_smoothed_per_million``` - Novos casos nos últimos 7 dias
- ```total_deaths_per_million``` - Número total de mortes a cada 1.000.000 de pessoas
- ```new_deaths_per_million``` - Novas mortes confirmadas a cada 1.000.000 de pessoas
- ```new_deaths_smoothed_per_million``` - Média de novas mortes nos últimos 7 dias
- ```reproduction_rate``` - Taxa de reprodução do COVID-19 (estimativa)
- ```icu_patients``` e ```icu_patients_per_million``` - Número de pacientes em UTIs
- ```hosp_patients```e ```hosp_patients_Per_million``` - Número de pacientes com COVID-19 nos hospitais no dia observado
- ```weekly_icu_admissions``` e ```weekly_icu_admissions_per_million``` - Número de pacientes entrantes em UTIs na última semana (dia atual + 6 dias anteriores)
- ```weekly_hosp_admissions``` e ```weekly_hosp_admissions_per_million``` - Número de pacientes entranes em hospitais na última semana (dia atual + 6 dias anteriores)
- ```total_tests``` - Número total de testes
- ```new_tests``` - Número de novos testes no dia
- ```total_tests_per_thousand``` - Número total de testes a cada 1000 pessoas
- ```new_tests_per_thousand```- Número de novos testes a cada 1000 pessoas
- ```new_tests_smoothed``` e ```new_tests_smoothed_per_thousand``` - Média dos últimos 7 dias de novos testes
- ```positive_rate``` - A porcentagem de teste de COVID-19 que são positivos (Média dos últimos 7 dias)
- ```tests_per_case``` - A cada quantos testes é detectado um caso de COVID-19 confirmado
- ```tests_units``` - Número de locais que disponibilizam os testes para a COVID-19, que forneceram os dados para a base
- ```total_vaccinations``` e ```total_vaccinations_per_hundred``` - Número total de vacinas aplicadas no país
- ```people_vaccinated``` e ```people_vaccinated_per_hundred``` - Número total de pessoas que receberam pelo menos uma dose
- ```people_fully_vaccinated``` e ```people_fully_vaccinated_per_hundred``` - Número total de pessoas que receberam todas as doses pre-escritas pelo protocólo inicial de vacinação
- ```total_boosters``` e ```total_boosters_per_hundred``` - Número total de doses de reforço administradas
- ```new_vaccinations``` , ```new_vaccinations_smoothed``` e ```new_vaccinations_smoothed_per_million``` - Novas doses administradas
- ```stringency_index``` - Resposta do governo a restrição (fechamento de escolas, escritórios, viajens e etc), variando de 0 a 100, 100 = maior restrição
- ```population_density``` - Número de pessoas, divido pela extensão territorial (medido em km²)
- ```median_age``` - Média da idade da população
- ```aged_65_older``` - Porcentagem da população a cima de 65 anos
- ```aged_70_older``` - Porcentagem da população a cima de 70 anos
- ```gdp_per_capita``` - Produto Interno Bruto (PIB) per Capita em dólares
- ```extreme_poverty``` - Porcentagem da população em extrema pobreza
- ```cardiovasc_death_rate``` - Porcentagem de mortes decorrentes de doenças cardio-vasculares em 2017 - (Número anual de mortes a cada 100.000 pessoas)
- ```diabetes_prevalence``` - Porcentagem da população com diabetes (dos 20 aos 79) em 2017
- ```female_smokers``` - Porcentagem de mulheres que fumam
- ```male_smokers``` - Porcentagem de homens que fumam
- ```handwashing_facilities``` - Porcentagem da população com saneamento básico
- ```hospital_beds_per_thousand``` - Camas de hospital a cada 1000 pessoas
- ```life_expectancy``` - Expectativa de vida
- ```human_development_index``` - Índice de desenvolvimento humano (Varia de 0 a 1, sendo 1 = máximo), analisa 3 pilares de um país: expectativa/qualidade de vida, Educação e qualidade de vida
- ```population``` - População atual do país
- ```excess_mortality_cumulative_absolute``` - Diferença absoluta entre o número real de mortes ocorridas e a previsão da mesma data de n° de mortos no ano anterior
- ```excess_mortality_cumulative``` e ```excess_mortality_cumulative_per_million``` - Diferença em porcentagem entre o número real de mortes ocorridas e a previsão da mesma data de n° de mortos no ano anterior

### Importando os dados do COVID-19

Os dados originais estão disponíveis do seguinte repositório do github: https://github.com/owid/covid-19-data/blob/master/public/data/owid-covid-data.csv

Contudo, a fim de garantir uma possível replicação e a perpetuação da base de dados, durante o estudo será utilizada a mesma base de dados em um link que o responsável por este estudo será o mantenedor.

[Link para o repositório](https://raw.githubusercontent.com/vinihora/portfolio_data_science/main/dados/owid-covid-data.csv)

In [145]:
# importar as bibliotecas necessárias
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np


# importar o dataset COVID-19 para um DataFrame
df = pd.read_csv('https://raw.githubusercontent.com/vinihora/portfolio_data_science/main/dados/owid-covid-data.csv', parse_dates=True)

## Classificando os países entre desenvolvidos e subdesenvolvidos

- A fim classificar os países entre os 2 diferentes grupos, adotaremos algumas premissas para considerá-los como desenvolvidos ou subdesenvolvidos a fim de atender a necessidade do estudo

- A primeira premissa utilizada no modelo será em relação ao **Índice de Desenvolvimento Humano (IDH)**, que está relacionado a variável ```humam_development_index```. Os países desenvolvidos terão de possuir um valor **acima de 0.9**

- Outra premissa utilizada será o PIB per capita superior a $30.000 por ano para ser considerado um país desenvolvido

## Aplicando as premissas
- Primeiro, vamos descobrir quais países atendem aos requisitos, com o código de bloco abaixo

In [146]:
# Localizando no DataFrame países que possuem um IDH > 0.9 e PIB per capita > 300000
developed_countries = df.loc[(df['human_development_index'] > 0.9) & (df['gdp_per_capita'] > 30000)].location.unique()
developed_countries

array(['Australia', 'Austria', 'Belgium', 'Canada', 'Denmark', 'Finland',
       'France', 'Germany', 'Hong Kong', 'Iceland', 'Ireland', 'Israel',
       'Japan', 'Luxembourg', 'Netherlands', 'New Zealand', 'Norway',
       'Singapore', 'Slovenia', 'South Korea', 'Spain', 'Sweden',
       'Switzerland', 'United Kingdom', 'United States'], dtype=object)

- No próximo bloco de código será adicionado uma nova coluna ao data frame com o nome ```developed```, sinalizando se o país está na lista de desenvolvidos ou não

In [147]:
# Criando a coluna, atribuindo todos os valores como False
df['developed'] = False

# Atribuindo o valor True para os países que se enquadram nas premissas adotadas
df.loc[(df['human_development_index'] > 0.9) & (df['gdp_per_capita'] > 30000), 'developed'] = True

- Verificado se a adição foi feita com sucesso

In [148]:
# Verificando as 5 primeiras observações de países desenvolvidos
df.loc[(df['developed'] == True)].head(5)

iso_code continent   location        date  total_cases  new_cases  \
16653      AUS   Oceania  Australia  2020-01-03          NaN        0.0   
16654      AUS   Oceania  Australia  2020-01-04          NaN        0.0   
16655      AUS   Oceania  Australia  2020-01-05          NaN        0.0   
16656      AUS   Oceania  Australia  2020-01-06          NaN        0.0   
16657      AUS   Oceania  Australia  2020-01-07          NaN        0.0   

       new_cases_smoothed  total_deaths  new_deaths  new_deaths_smoothed  \
16653                 NaN           NaN         0.0                  NaN   
16654                 NaN           NaN         0.0                  NaN   
16655                 NaN           NaN         0.0                  NaN   
16656                 NaN           NaN         0.0                  NaN   
16657                 NaN           NaN         0.0                  NaN   

       total_cases_per_million  new_cases_per_million  \
16653                      NaN                    0.0   
16654                      NaN                    0.0   
16655                      NaN                    0.0   
16656                      NaN                    0.0   
16657                      NaN                    0.0   

       new_cases_smoothed_per_million  total_deaths_per_million  \
16653                             NaN                       NaN   
16654                             NaN                       NaN   
16655                             NaN                       NaN   
16656                             NaN                       NaN   
16657                             NaN                       NaN   

       new_deaths_per_million  new_deaths_smoothed_per_million  \
16653                     0.0                              NaN   
16654                     0.0                              NaN   
16655                     0.0                              NaN   
16656                     0.0                              NaN   
16657                     0.0                              NaN   

       reproduction_rate  icu_patients  icu_patients_per_million  \
16653                NaN           NaN                       NaN   
16654                NaN           NaN                       NaN   
16655                NaN           NaN                       NaN   
16656                NaN           NaN                       NaN   
16657                NaN           NaN                       NaN   

       hosp_patients  hosp_patients_per_million  weekly_icu_admissions  \
16653            NaN                        NaN                    NaN   
16654            NaN                        NaN                    NaN   
16655            NaN                        NaN                    NaN   
16656            NaN                        NaN                    NaN   
16657            NaN                        NaN                    NaN   

       weekly_icu_admissions_per_million  weekly_hosp_admissions  \
16653                                NaN                     NaN   
16654                                NaN                     NaN   
16655                                NaN                     NaN   
16656                                NaN                     NaN   
16657                                NaN                     NaN   

       weekly_hosp_admissions_per_million  total_tests  new_tests  \
16653                                 NaN          NaN        NaN   
16654                                 NaN          NaN        NaN   
16655                                 NaN          NaN        NaN   
16656                                 NaN          NaN        NaN   
16657                                 NaN          NaN        NaN   

       total_tests_per_thousand  new_tests_per_thousand  new_tests_smoothed  \
16653                       NaN                     NaN                 NaN   
16654                       NaN                     NaN                 NaN   
16655                       NaN         

In [149]:
# Verificando as 5 primeiras observações de países não desenvolvidos
df.loc[(df['developed'] == False)].head(5)

iso_code continent     location        date  total_cases  new_cases  \
0      AFG      Asia  Afghanistan  2020-01-03          NaN        0.0   
1      AFG      Asia  Afghanistan  2020-01-04          NaN        0.0   
2      AFG      Asia  Afghanistan  2020-01-05          NaN        0.0   
3      AFG      Asia  Afghanistan  2020-01-06          NaN        0.0   
4      AFG      Asia  Afghanistan  2020-01-07          NaN        0.0   

   new_cases_smoothed  total_deaths  new_deaths  new_deaths_smoothed  \
0                 NaN           NaN         0.0                  NaN   
1                 NaN           NaN         0.0                  NaN   
2                 NaN           NaN         0.0                  NaN   
3                 NaN           NaN         0.0                  NaN   
4                 NaN           NaN         0.0                  NaN   

   total_cases_per_million  new_cases_per_million  \
0                      NaN                    0.0   
1                      NaN                    0.0   
2                      NaN                    0.0   
3                      NaN                    0.0   
4                      NaN                    0.0   

   new_cases_smoothed_per_million  total_deaths_per_million  \
0                             NaN                       NaN   
1                             NaN                       NaN   
2                             NaN                       NaN   
3                             NaN                       NaN   
4                             NaN                       NaN   

   new_deaths_per_million  new_deaths_smoothed_per_million  reproduction_rate  \
0                     0.0                              NaN                NaN   
1                     0.0                              NaN                NaN   
2                     0.0                              NaN                NaN   
3                     0.0                              NaN                NaN   
4                     0.0                              NaN                NaN   

   icu_patients  icu_patients_per_million  hosp_patients  \
0           NaN                       NaN            NaN   
1           NaN                       NaN            NaN   
2           NaN                       NaN            NaN   
3           NaN                       NaN            NaN   
4           NaN                       NaN            NaN   

   hosp_patients_per_million  weekly_icu_admissions  \
0                        NaN                    NaN   
1                        NaN                    NaN   
2                        NaN                    NaN   
3                        NaN                    NaN   
4                        NaN                    NaN   

   weekly_icu_admissions_per_million  weekly_hosp_admissions  \
0                                NaN                     NaN   
1                                NaN                     NaN   
2                                NaN                     NaN   
3                                NaN                     NaN   
4                                NaN                     NaN   

   weekly_hosp_admissions_per_million  total_tests  new_tests  \
0                                 NaN          NaN        NaN   
1                                 NaN          NaN        NaN   
2                                 NaN          NaN        NaN   
3                                 NaN          NaN        NaN   
4                                 NaN          NaN        NaN   

   total_tests_per_thousand  new_tests_per_thousand  new_tests_smoothed  \
0                       NaN                     NaN                 NaN   
1                       NaN                     NaN                 NaN   
2                       NaN                     NaN                 NaN   
3                       NaN                     NaN                 NaN   
4                       NaN                     NaN                 NaN   

   new_tests_smoothed_per_

- A partir das amostrar acima, podemos verificar que a nova coluna foi adicionada ao DataFrame e está de acordo com as regras pré-estabelecidas

# **2.Análise Exploratória dos Dados**

- Agora com a nova variável já criada, serão realizadas inúmeras análises com o intuito de entender quais serão os dados pertinentes, que nos levará a respostas de acordo com o nosso objetivo
- O conjunto de dados possuem diversas colunas com informações relevantes. Contudo, nossa análise será prioritariamente voltada a diferença e efeitos da COVID-19 entre os países desenvolvidos e subdesenvolvidos, visto isso, algumas das colunas serão retiradas da análise
- Além disso, serão observados os outliers presentes no dataset e serão tomadas as melhores decisões a fim de não prejudicar o andamento e resultados da análise
- Por fim, após os dados já tratados, as análises serão realizadas seguindo a trilha de resposta para o problema de pesquisa



## 2.1 Exploração inicial

- Primeiro, será decidido quais variáveis serão pertinentes ao estudo.
### Variáveis selecionadas:
* ```iso_code``` - facilita a identificação dos países
* ```continent``` - será útil para entender a distância do país em relação ao foco da doença
* ```location``` - país analisado
* ```date``` -
* ```total_cases``` - evolução dos casos
* ```new_cases```
* ```total_deaths```
* ```new_deaths```
* ```total_cases_per_million``` - facilita a comparação de casos de COVID-19 entre países com diferentes populações
* ```new_cases_per_million``` - facilita a comparação de mortes causadas pela COVID-10 entre países com diferente populações
* ```total_deaths_per_million```
* ```new_deaths_per_million```
* ```reproduction_rate``` - Comparação se as condições estabelecidas no país são relevantes para a reprodução do vírus
* ```total_tests```
* ```total_tests_per_thousand``` - Número total de testes por país, a cada 1000 habitantes
* ```new_tests_per_thousand``` - Novos testes realizados (Será que os testes aumentaram nos países subdesenvolvidos só depois dos casos em países desenvolvidos diminuírem? Sugerindo uma prioridade aos países desenvolvidos)
* ```positive_rate``` - Será que os países subdesenvolvidos tiveram um número maior de testes positivados?
* ```tests_per_case``` - Será que a média de testes por caso confirmado nos países subdesenvolvidos são menores do que em países desenvolvidos (Um número menor de testes por casos confirmados pode sugerir que só são realizados testes em indivíduos com sintomas extremamentes gritantes em países subdesenvolvidos)
* ```total_vaccinations``` -
* ```people_vaccinated``` -
* ```people_fully_vaccinated``` -
* ```total_boosters``` -
* ```new_vaccinations``` -
* ```total_vaccinations_per_hundred```
* ```people_vaccinated_per_hundred```
* ```total_boosters_per_hundred```
* ```stringency_index``` - Será que países com menores níveis de desenvolvimento não foram capazes de aplicar grandes restrições devido a necessidade da população de continuar em seus trabalhos e falta de infraestrutura para adotar o modelo híbrido em massa?
* ```population_density``` - Afeta o número de casos ou de taxa de reprodução?
* ```gdp_per_capita```
* ```handwashing_facilities```
* ```hospital_beds_per_thousand```
* ```developed```




In [150]:
# Mostra todas as colunas do DF (DataFrame)
pd.set_option('display.max_columns', None)

# Cria uma cópia para ser possível manipular o DF sem nenhum prejuízo do original
df_change = df.copy()

# Filtrando somente as colunas que serão utilizadas
df_change = df[['iso_code', 'continent', 'location', 'date', 'total_cases', 'new_cases', 'total_deaths', 'new_deaths', 'total_cases_per_million', 'new_cases_per_million', 'total_deaths_per_million', 'new_deaths_per_million', 'reproduction_rate', 'total_tests', 'total_tests_per_thousand', 'new_tests_per_thousand', 'positive_rate', 'tests_per_case', 'total_vaccinations', 'people_vaccinated', 'people_fully_vaccinated', 'total_boosters', 'new_vaccinations', 'total_vaccinations_per_hundred', 'people_vaccinated_per_hundred', 'total_boosters_per_hundred', 'stringency_index', 'population_density', 'gdp_per_capita', 'handwashing_facilities', 'hospital_beds_per_thousand', 'developed']]

# Visualizando o novo DF
df_change.head(5)

iso_code continent     location        date  total_cases  new_cases  \
0      AFG      Asia  Afghanistan  2020-01-03          NaN        0.0   
1      AFG      Asia  Afghanistan  2020-01-04          NaN        0.0   
2      AFG      Asia  Afghanistan  2020-01-05          NaN        0.0   
3      AFG      Asia  Afghanistan  2020-01-06          NaN        0.0   
4      AFG      Asia  Afghanistan  2020-01-07          NaN        0.0   

   total_deaths  new_deaths  total_cases_per_million  new_cases_per_million  \
0           NaN         0.0                      NaN                    0.0   
1           NaN         0.0                      NaN                    0.0   
2           NaN         0.0                      NaN                    0.0   
3           NaN         0.0                      NaN                    0.0   
4           NaN         0.0                      NaN                    0.0   

   total_deaths_per_million  new_deaths_per_million  reproduction_rate  \
0                       NaN                     0.0                NaN   
1                       NaN                     0.0                NaN   
2                       NaN                     0.0                NaN   
3                       NaN                     0.0                NaN   
4                       NaN                     0.0                NaN   

   total_tests  total_tests_per_thousand  new_tests_per_thousand  \
0          NaN                       NaN                     NaN   
1          NaN                       NaN                     NaN   
2          NaN                       NaN                     NaN   
3          NaN                       NaN                     NaN   
4          NaN                       NaN                     NaN   

   positive_rate  tests_per_case  total_vaccinations  people_vaccinated  \
0            NaN             NaN                 NaN                NaN   
1            NaN             NaN                 NaN                NaN   
2            NaN             NaN                 NaN                NaN   
3            NaN             NaN                 NaN                NaN   
4            NaN             NaN                 NaN                NaN   

   people_fully_vaccinated  total_boosters  new_vaccinations  \
0                      NaN             NaN               NaN   
1                      NaN             NaN               NaN   
2                      NaN             NaN               NaN   
3                      NaN             NaN               NaN   
4                      NaN             NaN               NaN   

   total_vaccinations_per_hundred  people_vaccinated_per_hundred  \
0                             NaN                            NaN   
1                             NaN                            NaN   
2                             NaN                            NaN   
3                             NaN                            NaN   
4                             NaN                            NaN   

   total_boosters_per_hundred  stringency_index  population_density  \
0                         NaN               0.0              54.422   
1                         NaN               0.0              54.422   
2                         NaN               0.0              54.422   
3                         NaN               0.0              54.422   
4                         NaN               0.0              54.422   

   gdp_per_capita  handwashing_facilities  hospital_beds_per_thousand  \
0        1803.987                  37.746                         0.5   
1        1803.987                  37.746                         0.5   
2        1803.987                  37.746                         0.5   
3        1803.987                  37.746                         0.5   
4        1803.987                  37.746                         0.5   

   developed  
0      False  
1      False  
2      False  
3      False  
4      False

- Pode-se perceber que a base de dados possue alguns valores nulos em algumas colunas, isso ocorre pois na data apresentada, ainda não haviam dados disponíveis para o país observado
- Sendo assim, será necessário realizar um tratamento em que remove todas as linhas em que as colunas total_cases == Nulo & new_cases == 0, pois assim garantiremos que só haverá observações em que o COVID esteve presente no país
- Antes de aplicar as alterações, vamos verificar o formato da base da dados (número de linhas x número de colunas)

In [151]:
df_change.shape

(323798, 32)

- O Dataset apresentou um total de **323798 observações**. Porém, nem todas as observações serão úteis para a nossa análise. Visto isso, vamos retirar àquelas que não é pertinente

In [152]:
# Aplicando as condições descritas a cima (Excluindo as as linhas que possuem as colunas total_cases como nulo e new_cases igual a 0)
df_change = df_change.loc[(df_change['total_cases'].notnull()) & (df_change['new_cases'] != 0)]


In [153]:
# Reiniciando a contagem das linhas a partir do índice
df_change = df_change.reset_index()
df_change.drop('index', axis=1, inplace=True)

df_change

iso_code continent     location        date  total_cases  new_cases  \
0           AFG      Asia  Afghanistan  2020-02-26          1.0        1.0   
1           AFG      Asia  Afghanistan  2020-03-09          4.0        3.0   
2           AFG      Asia  Afghanistan  2020-03-12          7.0        3.0   
3           AFG      Asia  Afghanistan  2020-03-16         10.0        3.0   
4           AFG      Asia  Afghanistan  2020-03-17         16.0        6.0   
...         ...       ...          ...         ...          ...        ...   
173221      ZWE    Africa     Zimbabwe  2023-06-05     265161.0        9.0   
173222      ZWE    Africa     Zimbabwe  2023-06-12     265289.0      128.0   
173223      ZWE    Africa     Zimbabwe  2023-06-19     265413.0      124.0   
173224      ZWE    Africa     Zimbabwe  2023-06-26     265524.0      111.0   
173225      ZWE    Africa     Zimbabwe  2023-07-03     265604.0       80.0   

        total_deaths  new_deaths  total_cases_per_million  \
0                NaN         0.0                    0.024   
1                NaN         0.0                    0.097   
2                NaN         0.0                    0.170   
3                NaN         0.0                    0.243   
4                NaN         0.0                    0.389   
...              ...         ...                      ...   
173221        5695.0         0.0                16247.074   
173222        5702.0         7.0                16254.917   
173223        5707.0         5.0                16262.514   
173224        5707.0         0.0                16269.316   
173225        5709.0         2.0                16274.217   

        new_cases_per_million  total_deaths_per_million  \
0                       0.024                       NaN   
1                       0.073                       NaN   
2                       0.073                       NaN   
3                       0.073                       NaN   
4                       0.146                       NaN   
...                       ...                       ...   
173221                  0.551                   348.947   
173222                  7.843                   349.376   
173223                  7.598                   349.682   
173224                  6.801                   349.682   
173225                  4.902                   349.805   

        new_deaths_per_million  reproduction_rate  total_tests  \
0                        0.000                NaN          NaN   
1                        0.000                NaN          NaN   
2                        0.000                NaN          NaN   
3                        0.000                NaN          NaN   
4                        0.000                NaN          NaN   
...                        ...                ...          ...   
173221                   0.000                NaN          NaN   
173222                   0.429                NaN          NaN   
173223                   0.306                NaN          NaN   
173224                   0.000                NaN          NaN   
173225                   0.123                NaN          NaN   

        total_tests_per_thousand  new_tests_per_thousand  positive_rate  \
0                            NaN                     NaN            NaN   
1                            NaN                     NaN            NaN   
2                            NaN                     NaN            NaN   
3                            NaN                     NaN            NaN   
4                            NaN                     NaN            NaN   
...                          ...                     ...            ...   
173221                       NaN                     NaN            NaN   
173222                       NaN                     NaN            NaN   
173223                       NaN                     NaN            NaN   
173224                       NaN                     NaN            NaN   
173225    

In [154]:
df_change.shape

(173226, 32)

- Após o filtro realizado, podemos observar uma queda enorme de linhas na base de dados (cerca de 53%)
- Apesar dessa queda no número de observações, agora é possível retirar insights das linhas que permanecerem, pois nos revelam alguma informação que utilizaremos na análise

### Análise Exploratória Inicial

* *Visualizar as 5 primeiras entradas*
    * `df.head()`
* *Qual o tamanho do seu DataFrame*
    * `df.shape`
* *Extrair e imprimir os nomes das colunas*
    * `df.columns`
* *Quais os tipos das variáveis*
    * `df.dtypes`
* *Identifique a porcentagem de valores ausentes das colunas*
    * `df.isnull().sum() / df.shape[0]` ou
    * `(df.isnull().sum() / df.shape[0]).sort_values(ascending=False)` *para ordem decrescente.*
* *Transforme a coluna `date` no formato `datetime`*
    * `df.date = pd.to_datetime(df.date)`
* *Ver para a data mais atual, quais os países que têm mais casos e mortes. Veja o exemplo:*
    * `df.loc[df.date == '2020-06-14'].sort_values(by="total_cases", ascending=False)`
    * `df.loc[df.date == '2020-06-14'].sort_values(by="total_deaths", ascending=False)`
* Plotar um gráfico de barras para os 5 países com mais mortes
    * `df.loc[df.date == '2020-06-14', ['location', 'total_deaths']].sort_values(by="total_deaths", ascending=False)[1:6]`
    * `ax.bar('location', 'total_deaths', data=top5_total_deaths)`
* Plotar um gráfico de linhas para ver a evolução do total de casos de COVID-19 no mundo.
    * `df.loc[df.location == "World", "total_deaths"].reset_index(drop=True).plot()`
* [OPCIONAL] Plotar um gráfico de dispersão considerando `gdp_per_capita` (ou seja, o PIB dos países) pelo número total de mortos. Pegue a data mais recente apenas.
    * Lembre-se que não é nada conclusivo, pois envolve fatores como proporcão de testes, etc.


Vamos......

### Análise Exploratória para o Brasil

* Criar uma cópia do DataFrame apenas com o Brasil
    * `df_brasil = df.loc[df.location == "Brazil"].copy()`
* Identificar quando foi registrada a primeira morte no Brasil
* Identificar quantos dias demorou para registrarmos a primeira morte, considerando o primeiro caso de COVID-19 registrado.
* Plotar gráficos de linha em escalas linear e logarítmica para os casos totais e mortes totais.

## Conclusão

*Escreva suas conclusões e análises sobre os dados relativos à COVID-19...*